File: python/while-loop.py
import sys
import random

running = True

while running:
	solution = random.randint(1, 100)
	correct = False

	print("You have 10 tries to guess my number between 1 and 100.")

	for i in range(10):
		print(f"{(i+1)/10} What's your guess?")
		guess = int(sys.stdin.readline())

		if guess < solution:
			print("My number is bigger.");
		elif guess > solution:
			print("My number is smaller.");
		else:
			correct = True
			print("That's it. You guessed it.");
			break

	if not correct:
		print("Ohh you didn't guessed it. :(")

	print("Do you want to play another round?")
	running = sys.stdin.readline().strip() == "yes"

File: cpp/while-loop.cpp
#include <iostream>
#include <string>
#include <time.h>
#include <stdlib.h>

using namespace std;

int main() {
	bool running = true;
	while (running) {
		srand (time(NULL));
		const int solution = rand() % 100 + 1;

		bool correct = false;

		cout << "You have 10 tries to guess my number between 1 and 100." << endl;

		for (int i = 1; i <= 10; i++) {
			cout << "[" << i << "/10] What's your guess?" << endl;

			string input;
			getline(cin, input);
			int guess = stoi(input);

			if (guess < solution) {
				cout << "My number is bigger." << endl;
			} else if (guess > solution) {
				cout << "My number is smaller." << endl;
			} else {
				correct = true;
				cout << "That's it. You guessed it." << endl;
				break;
			}
		}

		if (!correct) {
			cout << "Ohh you didn't guess it. :(" << endl;
		}

		cout << "Do you want to play another round?" << endl;
		string continueGame;
		getline(cin, continueGame);

		running = continueGame == "yes";
	} 
	
	return 0;
}